File: Data Structures/Array/dominantIndex.py
class Solution:
    def dominantIndex(self, arr: list[int]) -> int:
        m = max(arr)
        maxIndex = arr.index(m)

        for i, v in enumerate(arr):
            if m != v and m < 2 * v:
                maxIndex = -1
                break
        return maxIndex
        

File: Data Structures/Array/dominantIndex.go
package main

func dominantIndex(arr []int) int {
	maxIndex, m := 0, 0

	for index, value := range arr {
		if m < value {
			m = value
			maxIndex = index
		}
	}

	for _, value := range arr {
		if m != value && m < 2*value {
			maxIndex = -1
			break
		}
	}
	return maxIndex
}
